### PR TITLE
Centos6

### PR DESCRIPTION
**Pull request by raticate**
dockerfile centos6

### CODE REVIEW

File: tchsm/lib/centos6/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+RUN cd /tmp && mkdir libtc && cd libtc \
+ && git clone https://github.com/niclabs/tchsm-libtc.git \
+ && cd tchsm-libtc && mkdir build && cd build && cmake .. && make install
+
+
+RUN cd /tmp && mkdir libdtc && cd libdtc \
+ && git clone https://github.com/niclabs/tchsm-libdtc.git \
+ && cd tchsm-libdtc \
+# TODO TEMPORAL
+# && git checkout a9a3e299a46 \
```
Comment (fmontoto) [inline tchsm/lib/centos6/Dockerfile]: Add the checkout to v0.1
EDIT:
Actually, download the zip of the release (https://github.com/niclabs/tchsm-libdtc/archive/v0.1.zip) and do a hash check.
We should also have the same with libtc, can you do a release of it?

File: tchsm/lib/centos6/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+RUN curl -LO http://ufpr.dl.sourceforge.net/project/mhash/mhash/0.9.9.9/mhash-0.9.9.9.tar.gz \
+ && echo 'c898de5ea60d9e0873a1b73caa031bb1b5797c03  mhash-0.9.9.9.tar.gz' | sha1sum -c \
+ && tar -xzC /tmp -f mhash-0.9.9.9.tar.gz && rm mhash-0.9.9.9.tar.gz \
+ && cd /tmp/mhash-0.9.9.9 && ./configure && make && make install \
+ && cd / && rm -r /tmp/mhash-0.9.9.9
+
+RUN curl -LO http://www.hyperrealm.com/libconfig/libconfig-1.5.tar.gz \
+ && tar -xzC /tmp -f libconfig-1.5.tar.gz && rm libconfig-1.5.tar.gz \
+ && cd /tmp/libconfig-1.5 && ./configure && make install \
```
Comment (fmontoto) [inline tchsm/lib/centos6/Dockerfile:62]: Add a hash check here

File: tchsm/lib/centos6/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+ && cd /tmp/libsodium-1.0.9/ && ./configure && make && make install \
+ && cd / && rm -r /tmp/libsodium-1.0.9
+
+RUN curl -LO https://github.com/zeromq/zeromq4-1/releases/download/v4.1.4/zeromq-4.1.4.tar.gz \
+ && echo 'b632a4b6f8a14390dc17824e37ff7b10831ce2b4  zeromq-4.1.4.tar.gz' | sha1sum -c \
+ && tar -xzC /tmp -f zeromq-4.1.4.tar.gz && rm zeromq-4.1.4.tar.gz \
+ && cd /tmp/zeromq-4.1.4 && ./configure --with-libsodium && make && make install \
+ && ldconfig \
+ && cd / && rm -r /tmp/zeromq-4.1.4
+
```
Comment (fmontoto) [inline tchsm/lib/centos6/Dockerfile:45]: && cd / && rm -r /tmp/zeromq-4.1.4 \
&& rm -r /tmp/zeromq-4.1.4

**Comment by fmontoto** [comment]
Remove the compiler at the end of the Dockerfile and other installation only dependencies.

**Review by fmontoto — CHANGES_REQUESTED** [review]

File: tchsm/lib/centos6/Dockerfile (hunk 3 above)
Comment (raticate) [reply tchsm/lib/centos6/Dockerfile:45]: ??

Comment (raticate) [reply tchsm/lib/centos6/Dockerfile:45]: I really don't get this

**Review by fmontoto — APPROVED** [review]

**Pull request merged**